src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Dialog,
+  DialogTitle,
+  Button,
+  DialogContent,
+  IconButton,
+  Box,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { EmojiClickData } from "emoji-picker-react";
+import EmojiPickerIconButton from "../Common/EmojiPickerIconButton";
+import UserAvatar from "../Common/UserAvatar";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import formatTimestamp from "../../utilities/formatTimestamp";
+import axios from "axios";
+import { updatePost } from "../../state/slices/postsSlice";
+
+type EditPostModalProps = {
+  onClose: () => void;
+  openModal: boolean;
+  postId: number;
+  originalPostTextContent: string;
+  originalPostTimeStamp: string;
+  prevEditedPostTimestamp: string;
+};
+
+const styles = {
+  avatar: { paddingRight: 1.5 },
+  dialog: {
+    ".MuiDialog-scrollPaper": { alignItems: "flex-start" },
+  },
+  dialogContentContainer: {
+    padding: 2,
+  },
+  dialogTitle: {
+    display: "flex",
+    alignItems: "center",
+  },
+  editPostHeader: {
+    display: "flex",
+  },
+  paperProps: {
+    overflow: "visible",
+    borderRadius: 20,
+  },
+  postActions: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginTop: 1,
+  },
+  textFieldContainer: { width: "100%", paddingTop: 2 },
+  textField: { paddingBottom: 2 },
+  titleText: {
+    fontWeight: "bold",
+    paddingLeft: 3,
+  },
+  userInfoText: {
+    display: "flex",
+    gap: 0.5,
+  },
+};
+
+const EditPostModal = ({
+  onClose,
+  openModal,
+  postId,
+  originalPostTextContent,
+  originalPostTimeStamp,
+  prevEditedPostTimestamp,
+}: EditPostModalProps) => {
+  const [postTextContent, setPostTextContent] = useState("");
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const existingPost = useAppSelector((state) =>
+    state.posts.posts.find((post) => post.postId === postId)
+  );
+
+  useEffect(() => {
+    setPostTextContent(originalPostTextContent);
+  }, [originalPostTextContent]);
+
+  const handleEdit = async () => {
+    try {
+      await axios.put(
+        "http://localhost:3001/api/posts/updatePost",
+        {
+          postId: postId,
+          textContent: postTextContent,
+          editedTimestamp: new Date(),
+        },
+        {
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+
+      if (existingPost) {
```

Comment:
The `existingPost` object is typed as `string | undefined` because of the `.find()` operation, but in reality we know it should never be undefined. The post id doesn't come from a user input, it comes from our code, so we should always get a post back

Either way, I think we can handle this situation more gracefully than this. I'd propose we either use a type assertion using the non-null assertion operator (`!`) or throw an error inside here. @NasihNazeem do you have any thoughts?

Reply:
Ive implemented both just to be safe. Can be overkill but cant be too safe